pyrealm/constants/pmodel_const.py

```diff
@@ -124,6 +130,9 @@ class PModelConst(ConstantsClass):
     k_CtoK: float = 273.15
     """Conversion from °C to K   (:math:`CtoK` , 273.15, -)"""
 
+    water_density_method: str = "fisher"
```

Comment:
Rather than setting the method here in the class definition (and changing the definition when you want to change just this parameter), it would be better to give it as an argument to the constructor.

Reply:
I don't follow. The current setup is that you'd switch using:

`calc_density_h20(tc, patm)`

to

`calc_density_h20(tc, patm, const=PModelConst(water_density_method="fisher")`

The reason for packaging those constant settings is that it is then easier to pass around a single set of shared settings, but I agree that the signatures of some functions are then a bit odd.

Reply:
Apologies, my fault, I misread what you are doing.